src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.cpp

```diff
@@ -0,0 +1,443 @@
+                                                  ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+                                                  return CHIP_NO_ERROR;
+                                              });
+    }
+    else
+    {
+        result = mDelegate.FindRootCert(ctx.mRequestPath.mEndpointId, ctx.mCommandHandler.GetAccessingFabricIndex(),
+                                        req.caid.Value(), [&](auto & certificate) -> CHIP_ERROR {
+                                            FindRootCertificateResponse::Type response;
+                                            DataModel::List<const TLSCertStruct::Type> details(&certificate, 1);
+                                            response.certificateDetails = details;
+                                            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+                                            return CHIP_NO_ERROR;
+                                        });
+    }
+    if (result == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (result != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    }
+}
+
+void TlsCertificateManagementServer::HandleLookupRootCertificate(HandlerContext & ctx,
+                                                                 const LookupRootCertificate::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsCertificateManagement: LookupRootCertificate");
+    auto result = mDelegate.LookupRootCertByFingerprint(ctx.mRequestPath.mEndpointId, ctx.mCommandHandler.GetAccessingFabricIndex(),
+                                                        req.fingerprint, [&](auto & certificate) -> CHIP_ERROR {
+                                                            LookupRootCertificateResponse::Type response;
+                                                            response.caid = certificate.caid;
+                                                            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+                                                            return CHIP_NO_ERROR;
+                                                        });
+    if (result == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (result != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    }
+}
+
+void TlsCertificateManagementServer::HandleRemoveRootCertificate(HandlerContext & ctx,
+                                                                 const RemoveRootCertificate::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsCertificateManagement: RemoveRootCertificate");
+    auto result = mDelegate.RemoveRootCert(ctx.mRequestPath.mEndpointId, ctx.mCommandHandler.GetAccessingFabricIndex(), req.caid);
```

Comment:
So elsewhere we have range checks on incoming cert ids but not here?

Reply:
Fixed, apologies (I think this came out of us realizing during review that we wanted to treat this differently & not return `NOT_FOUND`, even though spec doesn't say anything about it)